src/aap_eda/services/activation/manager.py

```diff
@@ -228,9 +228,13 @@ def _check_latest_instance_and_pod_id(self) -> None:
             raise exceptions.ActivationInstancePodIdNotFound(msg)
 
     def _check_non_finalized_instances(self) -> None:
-        instances = models.RulebookProcess.objects.filter(
-            activation=self.db_instance,
-        )
+        args = {"parent_type": self.db_instance_type}
+        if self.db_instance_type == ProcessParentType.EVENT_STREAM:
+            args["event_stream_id"] = self.db_instance.id
+        else:
+            args["activation_id"] = self.db_instance.id
```

Comment:
This code and the similar in https://github.com/ansible/eda-server/pull/657/files#diff-767f95d0b4918e197417276be62147629e68416233ba137c48b8c171b570f73cR1054
can simpler if you directly use `self.db_instance_type`. The value of the enum already matches the name of the field. Not even you need to pass the parent_type. 
```
args = {f"{self.db_instance_type}": self.db_instance}
```
We don't want to fill the code with these conditionals and we don't need it.